019-quarkus-consul/src/test/java/io/quarkus/qe/containers/ConsulContainer.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.qe.containers;
+
+import java.time.Duration;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+
+public class ConsulContainer extends GenericContainer<ConsulContainer> {
+    private static final int PORT = 8500;
+
+    public ConsulContainer() {
+        super("quay.io/bitnami/consul:1.9.3");
+        waitingFor(new LogMessageWaitStrategy().withRegEx(".*Synced node info.*\\s"));
+        withStartupTimeout(Duration.ofMillis(60000));
+        addFixedExposedPort(PORT, PORT);
```

Comment:
I had to create the ConsulContainer because I can't inject a random port into the running test (the `inject` method is not called when the test is not annotated with `@QuarkusTest`: I will see if this is something we can fix).